### DISCUSSION TITLE
New optionswindow: wo_transparentbackground

### DISCUSSION THREAD

**Discussion by fredvs** [mse-org/mseide-msegui#52] [General]
Hello.

There is a new shape in optionswindow: _**wo_transparentbackground**_.
Setting this option allows to only see the child-components of the form, the background will be transparent.
It works for Linux and Windows.

You will need to recompile MSEide (or ideU) with the last commits of MSEgui to see that new option in the project inspector.

To enable it, just add in _form.oncreate_ this:

```
procedure tmainfo.oncreate(const sender: TObject);
begin
 optionswindow := [wo_groupleader,wo_taskbar,wo_alwaysontop,wo_transparentbackground]; // this as example
 SetChildBounds(sender); // this will create a list of child-bounds needed for creation of the shape of form.
end;
```

There is a demo in MSEuniverse: 
https://github.com/mse-org/mseuniverse/tree/main/samples/forms/transparentbackground

Here the form on the project:
![transpback_project](https://user-images.githubusercontent.com/3421249/174318656-40ca162b-a7cc-441c-a855-53d1a561cdc8.png)

And here compiled:
![transpback_compiled](https://user-images.githubusercontent.com/3421249/174318734-02228e01-df0b-43e8-b410-90f98390892f.png)

Fre;D

**Comment by fredvs** [comment]
Added in commit: [0943f1a](https://github.com/mse-org/mseide-msegui/commit/0943f1ae7a28e90c02aec02acedb9dc193c4236e) new window options:

_wo_transparentbackgroundround_: to have a transparent form with all the children visible and shape rounded.
You may use _mse_radiuscorner_ to set the radius of the rounded corners.

_wo_inalldesktops_: to have the form visible on all the desktops.

It works for X11 and Windows.

**Comment by fredvs** [comment]
Demo of the different window options:

With _default_ options:
![mseform_default](https://user-images.githubusercontent.com/3421249/174911318-10f59062-8696-4154-9fee-494da1f9d349.png)

With option _wo_noframe_
![mseform_rectangle](https://user-images.githubusercontent.com/3421249/174911442-17fb88a4-5a74-48d2-b848-502b3b2c2b55.png)

With option _wo_rounded_
![mseform_rounded](https://user-images.githubusercontent.com/3421249/174911502-60fac7da-d83a-448c-b8d9-139d3c6ad8d9.png)

With option _wo_ellipse_
![mseform_ellipse](https://user-images.githubusercontent.com/3421249/174911649-9a07708b-db20-40e1-ade1-b86ad2046c28.png)

With option _wo_transparentbackground_
![mseform_bkgrtrans](https://user-images.githubusercontent.com/3421249/174911777-ceb1aa51-b695-4243-8d4e-5b9a4e24478d.png)

With option _wo_transparentbackgroundround_
![mseform_bkgrtransround](https://user-images.githubusercontent.com/3421249/174911831-0132465f-a184-4a4f-a5f8-40f48379e1f5.png)

With option _wo_transparentbackgroundellipse_
![mseform_bkgrtranspellipse](https://user-images.githubusercontent.com/3421249/174911879-6324f1fb-e656-4bdf-b884-a70a403d1b8a.png)

**Comment by rchastain** [comment]
Good job Fred!
By the way, I don't find _wo_rounded_ in my documentation. I don't know why.
